src/main/scala/reporting/Converter.scala

```diff
@@ -0,0 +1,506 @@
+  override lazy val toString: String = value.toString
+  lazy val asValueEntry = {
+    if (value < 0.0) {
+      ApplicationEntry("-", Seq(ConstantEntry((-value).toString)))
+    } else {
+      ConstantEntry(value.toString)
+    }
+  }
+}
+
+case class RefEntry(
+    name: String,
+    fields: Map[String, (ExtractedModelEntry, Option[Rational])]
+) extends ExtractedModelEntry {
+  override lazy val toString: String = {
+    val buf = fields
+      .map(x =>
+        s"\t${x._1}(perm: ${x._2._2.map(_.toString).getOrElse("?")}) <- ${x._2._1.toString.split("\n").mkString("\n\t")}\n"
+      )
+      .mkString("")
+    s"Ref ($name) {\n$buf}"
+  }
+  lazy val asValueEntry = ConstantEntry(name)
+}
+
+case class NullRefEntry(name: String) extends ExtractedModelEntry {
+  override lazy val toString = s"Null($name)"
+  lazy val asValueEntry = ConstantEntry(name)
+}
+
+case class RecursiveRefEntry(name: String) extends ExtractedModelEntry {
+  override lazy val toString = s"recursive reference to $name"
+  lazy val asValueEntry = ConstantEntry(name)
+}
+
+case class VarEntry(name: String, sort: Sort) extends ExtractedModelEntry {
+  override lazy val toString: String = name
+  lazy val asValueEntry = ConstantEntry(name)
+}
+
+case class OtherEntry(value: String, problem: String = "")
+    extends ExtractedModelEntry {
+  override lazy val toString = s"$value [$problem]"
+  lazy val asValueEntry = ConstantEntry(value)
+}
+
+case class SeqEntry(name: String, values: Vector[ExtractedModelEntry])
+    extends ExtractedModelEntry {
+  override lazy val toString = s"($name): [${values.map(_.toString).mkString(", ")}]"
+  lazy val asValueEntry = ConstantEntry(name)
+}
+
+case class UnprocessedModelEntry(entry: ValueEntry)
+    extends ExtractedModelEntry {
+  override lazy val toString = s"$entry"
+  lazy val asValueEntry = entry
+}
+
+// processed Heap representation:
+sealed trait HeapEntry {
+  def toString: String
+}
+
+case class ExtractedHeap(entries: Vector[HeapEntry])
+
+case class PredHeapEntry(
+    name: String,
+    args: Seq[ExtractedModelEntry]
+) extends HeapEntry {
+  override lazy val toString = s"$name(${args.mkString(", ")})"
+}
+case class FieldHeapEntry(
+    recv: VarEntry,
+    field: String,
+    perm: Option[Rational],
+    sort: Sort,
+    entry: ExtractedModelEntry
+) extends HeapEntry {
+  override lazy val toString = s"$recv.$field"
+}
+
+case class UnresolvedHeapEntry(chunk: Chunk, reason: String) extends HeapEntry {
+  override lazy val toString = s"$chunk (not further processed because: $reason)"
+}
+
+object Converter {
+  lazy val termconverter = {
+    val conv = new TermToSMTLib2Converter()
+    conv.start()
+    conv
+  }
+  //some tokens used for naming model entries in a more maintainable way
+  lazy val seqLengthId: String = {
+    val term = SeqLength(Var(Identifier("any"), sorts.Seq(sorts.Int)))
+    termconverter.convert(term).split(" ").head.substring(1)
+  }
```

Comment:
@mschwerhoff so this is one example of how I used TermToSMTLib2Converter.convert() to get the String literals in a more maintainable way, but as you see from this example, it was not always possible for me to do it in a 100% maintainable way since I only use part of the returned String. So if just the names change, in this case Seq_length, this will still work but if the general structure changes there will still be a problem. Is this what you expected or do you think there is a better way to do this?

Reply:
Thanks for the example! The code obscures your actual intention so much, that it is probably better to go with string literals for now — with them, it is at least obvious what the purpose of the code is.

A better way will probably require changes to Silicon's term converter, e.g. add public vals that hold the relevant sort and symbol names, which seems to go beyond the scope of your thesis.

So, just leave the code as-is, and add a TODO comment for how to improve the code.

Reply:
Okay thanks, then I think I am pretty much done with the changes you suggested if you agree with them. 
But I struggle with the benchmarking for the silicon test suite. If I run your script on the silver tests it seems to produce something reasonable, but for silicon, even with the current master version, it states "The following output occurred during testing, but should not have according to the test annotations" or throws ParseErrors for almost every testcase. 
I tried both 
  ./utils/scripts/benchmark.sh src/test
  ./utils/scripts/benchmark.sh src/test/resources
the same commands with silvers test directory as target work fine. I made sure I have the most up to date version of both silver and silicon.

Reply:
Not all tests in Silicon's `src/test/resources` are meant to be verified stand-alone, maybe that explains it. Due to the upcoming semester start I unfortunately can't help you with debugging, but if the Silver test suite runs fine and doesn't reveal any significant performance differences, that should be good enough.